python/pyspark/pandas/tests/test_series.py

```diff
@@ -2071,6 +2071,22 @@ def test_asof(self):
         with ps.option_context("compute.eager_check", False):
             self.assert_eq(psser.asof(20), 4.0)
 
+        pser = pd.Series([2, 1, np.nan, 4], index=[10, 20, 30, 40], name="Koalas")
+        psser = ps.from_pandas(pser)
+        self.assert_eq(psser.asof([5, 20]), pser.asof([5, 20]))
+
+        pser = pd.Series([4, np.nan, np.nan, 2], index=[10, 20, 30, 40], name="Koalas")
+        psser = ps.from_pandas(pser)
+        self.assert_eq(psser.asof([5, 100]), pser.asof([5, 100]))
+
+        pser = pd.Series([np.nan, 4, 1, 2], index=[10, 20, 30, 40], name="Koalas")
+        psser = ps.from_pandas(pser)
+        self.assert_eq(psser.asof([5, 35]), pser.asof([5, 35]))
+
+        pser = pd.Series([2, 1, np.nan, 4], index=[10, 20, 30, 40], name="Koalas")
+        psser = ps.from_pandas(pser)
+        self.assert_eq(psser.asof([25, 25]), pser.asof([25, 25]))
```

Comment:
Can we also test the string & timestamp index ??

e.g.
```python
>>> pser = pd.Series([2, 1, np.nan, 4], index=['a', 'b', 'c', 'd'])
>>> pser.asof(['a', 'd'])
a    2.0
d    4.0
dtype: float64
```

```python
>>> pser = pd.Series([2, 1, np.nan, 4], index=[pd.Timestamp(2020, 1, 1), pd.Timestamp(2020, 2, 2), pd.Timestamp(2020, 3, 3), pd.Timestamp(2020, 4, 4)])
>>> pser.asof([pd.Timestamp(2020, 1, 1), pd.Timestamp(2020, 2, 4)])
2020-01-01    2.0
2020-02-04    1.0
dtype: float64
```

Reply:
For the timestamp , its throwing exception , since Timestamp name is  like tuple so pd.Index , throws assertion error  assert is_name_like_tuple(column_label, check_type=True), column_label .  However the earlier  code(suggested one)  is working fine  . Currently working on resolving it

Reply:
Had to revert to earlier code because of this issue .